lib/ex_machina/ecto_strategy.ex

```diff
@@ -23,4 +25,71 @@ defmodule ExMachina.EctoStrategy do
   def handle_insert(_record, _opts) do
     raise "expected :repo to be given to ExMachina.EctoStrategy"
   end
+
+  defp cast(record) do
+    record
+    |> cast_fields
+    |> cast_assocs
+  end
+
+  defp cast_fields(struct) do
+    struct
+    |> ExMachina.Ecto.drop_ecto_fields
+    |> Map.keys
+    |> cast_fields(struct)
+  end
+
+  defp cast_fields(fields, struct) do
+    Enum.reduce(fields, struct, fn(field, struct) ->
+      casted_value = cast_field(field, struct)
+      Map.put(struct, field, casted_value)
+    end)
+  end
+
+  defp cast_field(field, %{__struct__: schema} = struct) do
+    field_type = schema.__schema__(:type, field)
+    value = Map.get(struct, field)
+
+    if field_type do
```

Comment:
I'm not totally sure I understand this. Would the field ever not have a type? If so, maybe it would help to name this somehow to clarify why?